query/src/main/java/org/infinispan/query/impl/QueryDefinition.java

```diff
@@ -104,6 +111,10 @@ public SearchQueryBuilder getSearchQueryBuilder() {
       return searchQueryBuilder;
    }
 
+   public boolean maxResultsHasChanged() {
```

Comment:
I don't like the name. Maybe `isCustomMaxResults()` ?

Reply:
good idea. changed it